### ISSUE TITLE
Build failure with latest 8.4 patches - through a541ce9

### ISSUE THREAD

**Issue by ddssff** [ghcjs/ghcjs#708]
The build goes until it is building base, and then on the module Text.Show I get

```
[ 70 of 227] Compiling Data.Ord         ( Data/Ord.hs, dist/build/Data/Ord.js_o )
[ 71 of 227] Compiling Data.Either      ( Data/Either.hs, dist/build/Data/Either.js_o )
[ 72 of 227] Compiling Text.Read        ( Text/Read.hs, dist/build/Text/Read.js_o )
[ 73 of 227] Compiling Data.Char        ( Data/Char.hs, dist/build/Data/Char.js_o )
[ 74 of 227] Compiling Data.OldList     ( Data/OldList.hs, dist/build/Data/OldList.js_o )
[ 75 of 227] Compiling GHC.OldList      ( GHC/OldList.hs, dist/build/GHC/OldList.js_o )
[ 76 of 227] Compiling Text.Show        ( Text/Show.hs, dist/build/Text/Sho     2> panic! (the 'impossible' happened)
  (GHCJS version 8.4.0.1, GHC version 8.4.2.20180505):
        runtimeRepPrimRep
  typePrimRep (b_i3uEQ :: TYPE r2_i3uEO)
  r2_i3uEO
  Call stack:
      CallStack (from HasCallStack):
        callStackDoc, called at compiler/utils/Outputable.hs:1150:37 in ghc-api-ghcjs-8.4.0-inplace:Outputable
        pprPanic, called at compiler/simplStg/RepType.hs:358:5 in ghc-api-ghcjs-8.4.0-inplace:RepType

Please report this as a GHCJS bug:  https://github.com/ghcjs/ghcjs/issues

cabal: Error: some packages failed to install:
base-4.11.1.0-5Kf3GqzZBK3J87Uu7pRIno failed during the building phase. The
exception was:
ExitFailure 1
transformers-0.5.5.0-8ZjexgxxdBLH8OWn8CZkwx depends on transformers-0.5.5.0
which failed to install.

ghcjs-boot: error running: /usr/bin/cabal --config-file /usr/lib/ghcjs/.ghcjs/x86_64-linux-8.4.0.1-8.4.2.20180505/ghcjs/cabalBootConfig --ignore-sandbox install ./pkg/base ./pkg/integer-gmp ./pkg/transformers --allow-boot-library-installs --global --ghcjs --one-shot --avoid-reinstalls --builddir dist --with-compiler /usr/bin/ghcjs --with-hc-pkg /usr/bin/ghcjs-pkg --prefix /usr/lib/ghcjs/.ghcjs/x86_64-linux-8.4.0.1-8.4.2.20180505/ghcjs --enable-documentation --haddock-html --haddock-hoogle --haddock-hyperlink-source --enable-debug-info --disable-library-stripping --disable-executable-stripping -fghci --enable-profiling --enable-shared
exit status: 1
```

I've tried this three times with the same result.  OS is Ubuntu Bionic.

**Comment by ddssff** [comment]
I've run the library build more manually and it appears to actually die when trying to compile Data.Dynamic.

**Comment by ddssff** [comment]
Even more specifically, it is the function dynApply that causes the crash:

```
-- (f::(a->b)) `dynApply` (x::a) = (f a)::b
dynApply :: Dynamic -> Dynamic -> Maybe Dynamic
dynApply (Dynamic (Fun ta tr) f) (Dynamic ta' x)
  | Just HRefl <- ta `eqTypeRep` ta'
  , Just HRefl <- typeRep @Type `eqTypeRep` typeRepKind tr
  = Just (Dynamic tr (f x))
dynApply _ _
  = Nothing
```

**Comment by ddssff** [comment]
Build succeeds without the patches added this month.

**Comment by luite** [comment]
hmm, I missed a few lines in `base.cabal` in the update, but apparently that's not the problem. Now I wonder what I did in my tree to make this build...

At least it's easy to reproduce. I'll look into it tomorrow.

**Comment by luite** [comment]
1a0a47e385293261d92484312851ff1b9d845f4e makes it not fail. I haven't really tested it yet, but I suppose it was more broken before.

**Comment by ddssff** [comment]
It got further this time.  It died building ghcjs-th, possibly while running haddock?  I need to run it again to pinpoint the error.

**Comment by ddssff** [comment]
```
Configuring ghcjs-th-0.1.0.0...
Preprocessing library for ghcjs-th-0.1.0.0..
Building library for ghcjs-th-0.1.0.0..
[1 of 3] Compiling GHCJS.Prim.TH.Serialized ( GHCJS/Prim/TH/Serialized.hs, dist/build/GHCJS/Prim/TH/Serialized.js_o )
[2 of 3] Compiling GHCJS.Prim.TH.Types ( GHCJS/Prim/TH/Types.hs, dist/build/GHCJS/Prim/TH/Types.js_o )
panic! (the 'impossible' happened)
  (GHCJS version 8.4.0.1, GHC version 8.4.2.20180505):
        genCase: expression was not inline:
StgApp
( ghc-prim:GHC.CString.unpackCString#[Hr2lheFcCA2]G [StgLitArg MachStr "
)
is outside of enumeration's range
( 0,"]

h$r3 = h$$ghcjszmthzm0zi1zi0zi0zmGol2HMsP0B13xQ8dJyrA4QZCGHCJSziPrimziTHziTypes_b0;
h$r2 = h$$ghcjszmthzm0zi1zi0zi0zmGol2HMsP0B13xQ8dJyrA4QZCGHCJSziPrimziTHziTypes_bZ;
h$r1 = h$ghczmprimZCGHCziCStringziunpackCStringzh;
return h$ap_1_2_fast();


Please report this as a GHCJS bug:  https://github.com/ghcjs/ghcjs/issues

Failed to install ghcjs-th-0.1.0.0
cabal: Error: some packages failed to install:
ghcjs-th-0.1.0.0-Gol2HMsP0B13xQ8dJyrA4Q failed during the building phase. The
exception was:
ExitFailure 1
```

**Comment by ddssff** [comment]
@luite, wanted to make sure you noticed this new failure.

**Comment by luite** [comment]
Yes, fixed by 9d79b58

**Comment by ddssff** [comment]
Thank you!
